plone/app/contenttypes/migration/migration.py

```diff
@@ -205,10 +205,15 @@ def last_migrate_layout(self):
         migrate_properties.
         """
         old_layout = self.old.getLayout() or getattr(self.old, 'layout', None)
-        if old_layout in LISTING_VIEW_MAPPING.keys():
+        if old_layout:
             default_page = self.old.getDefaultPage() or \
                 getattr(self.old, 'default_page', None)
-            self.new.setLayout(LISTING_VIEW_MAPPING[old_layout])
+            try:
+                # Delete old-style layout attribute.
+                del self.new.layout
```

Comment:
On some content objects there is a layout attribute. That can differ from what context.getLayout() returns. I just delete it.
AFAIK that's old style. I don't know where the layout is stored new-style though....